### PR TITLE
Adds ability to build from branch on github

### PR DESCRIPTION
**Pull request by el-mapache**
Logic works but is a work-in-progress

### CODE REVIEW

**Comment by el-mapache** [comment]
I have to fix/add a test but this is ready to be reviewed!

File: api/authorizers/build.js [hunk 1]
```diff
@@ -1,47 +1,76 @@
...
+
+    if (build) {
+      return build;
+    }
+
+    return rejectBuild();
+  });
 };
 
 const getBuildByBranch = (user, params) => {
```
Comment (jseppi) [inline api/authorizers/build.js]: Kind of a bigger picture thought: I think the original intent for our `authorizer`s was to just check that a user had access to a specific resource. Seems like we've been leaking a bit more than that (like fetching objects) into the authorizers. We might want to have an architectural review of the approach. I think I'll file a separate `debt` issue for that discussion/evaluation.

Comment (jseppi) [reply api/authorizers/build.js]: Filed https://github.com/18F/federalist/issues/1497

Comment (el-mapache) [reply api/authorizers/build.js]: I agree, for me it starts to get tricky when the associations are eagerly loaded, it's not clear which resource is the one you don't have access to (or doesn't exist, in this case)

Maybe it's clearer in the long run to just perform the additional DB lookups?

**Pull request merged**